external/os-guides/pentesting/kali.md

```diff
@@ -6,595 +6,156 @@ redirect_from:
+In order to obtain a trusted fingerprint, check its value against multiple sources.
+Then, check the keys you download against your trusted fingerprint.
 
-This step is required since by (security) default a TemplateVM do not have a
+This step is required since by (security) default TemplateVM do not have a
 direct Internet connectivity. Users understanding the risks of enabling such
 access can change this configuration in firewall settings for the TemplateVM.
 
-**Note:** The prompt on each line indicates where each command should be entered
-(`@dom0`, `@kali-rolling`, `@xxxx-dvm` or `@debian-<X>`).
+1. Retrieve the Kali Linux PGP key using a DisposableVM.
 
-1. Retrive the Kali Linux GPG key using a DispVM.
+```shell_session
+$ gpg --keyserver hkps://keys.gnupg.net --recv-key 44C6513A8E4FB3D30875F758ED444FF07D8D0BF6
+$ gpg --list-keys --with-fingerprint 44C6513A8E4FB3D30875F758ED444FF07D8D0BF6 
+$ gpg --export --armor 44C6513A8E4FB3D30875F758ED444FF07D8D0BF6 > kali-key.asc
+```
 
-        [user@xxxx-dvm ~]$ gpg --keyserver hkp://keys.gnupg.net --recv-key 44C6513A8E4FB3D30875F758ED444FF07D8D0BF6
-        [user@xxxx-dvm ~]$ gpg --list-keys --with-fingerprint 44C6513A8E4FB3D30875F758ED444FF07D8D0BF6 
-        [user@xxxx-dvm ~]$ gpg --export --armor 44C6513A8E4FB3D30875F758ED444FF07D8D0BF6 > kali-key.asc
-
-2. **DO NOT TURN OFF** the DispVM, the `kali-key.asc` file will be copied to
-   the Kali Linux template in a further step.
+2. **DO NOT TURN OFF** the DisposableVM, the `kali-key.asc` file will be copied in
+   the Kali Linux template for a further step.
 
 3. Make sure the key is the authentic Kali key.
    See the [Kali website] for further advice and instructions on verification.
 
-#### Create a Kali Linux (rolling) template
-
+Create a Kali Linux (rolling) template
+----------------------------------------
 These instructions will show you how to upgrade a Debian TemplateVM to Kali Linux.
 
 1. (Optional) Check for latest Debian stable template and install it (if not already done)
 
-        [user@dom0 ~]$ sudo qubes-dom0-update --action="search all" qubes-template-debian
-        [user@dom0 ~]$ sudo qubes-dom0-update <latest Debian template>
+```shell_session
+# qubes-dom0-update --action="search all" qubes-template-debian
+# qubes-dom0-update <latest Debian template>
+```
 
-2. Start, update and close your latest Debian template
+2. Clone `debian-X` template
 
-        [user@dom0 ~]$ qvm-start debian-<X>
-        [user@dom0 ~]$ qvm-run -a debian-<X> gnome-terminal
-        [user@debian-<X> ~]$ sudo apt-get update
-        [user@debian-<X> ~]$ sudo apt-get upgrade
-        [user@dom0 ~]$ qvm-shutdown debian-<X>
+```shell_session
+$ qvm-clone debian-<X> kali-rolling
+```
 
-3. Clone `debian-X` template
+3. Check the name of currently used repository in `/etc/apt/sources.list.d/qubes-r<X>.list` and current testing [Debian release][Debian-releases]. Update repository list accordingly
 
```

Comment:
Add warning about any changes in Debian repository naming conventions.
For example in change to bullseye you have to take account of the change to bullseye-security which your `sed` example wont cover

Reply:
I completely remove every link to any Debian official repository and replace it with Kali repository, so no issue with naming convention. The only remaining thing related to `buster` / `bullseye` is Qubes OS repository